### ISSUE TITLE
Machine Learning for Biology (ml4bio)

### ISSUE THREAD

**Issue by agitter** [carpentries-incubator/proposals#110]
Thank you for your interest in developing and sharing lesson materials! To submit lesson materials or suggest a topic for future curricular development, please answer the questions below. Our Curriculum Development Team will follow up to suggest next steps in your lesson's trajectory. Questions? Please email team@carpentries.org.

1. What is the topic of your lesson or lesson proposal?

> The Machine Learning for Biology (ml4bio) workshop is aimed at biologists with no previous machine learning experience. This is a conceptual workshop designed to teach machine learning literacy. After the workshop, the participants will be able to define machine learning concepts like samples, features, training set, validation set, test set, evaluation metrics, and model selection. The participants will be able to interactively develop an understanding of machine learning classifiers commonly used in biology like decision trees, random forests, logistic regression, and neural networks. The focus will be on problems in biology where machine learning is effectively used.

2. Do you already have a draft of your lesson? You're welcome to share materials at any stage of development. If you already have drafted materials, please include a link.

- https://github.com/gitter-lab/ml-bio-workshop
- https://gitter-lab.github.io/ml-bio-workshop/

(If you answered "No" to question 2, you can skip the remaining questions. Thank you for your lesson idea!)

3. Do your materials conform to our [Code of Conduct][coc]?

Yes

4. Are your materials already on GitHub and do they use The Carpentries [lesson template][lesson-template]? (you can visit [our lesson example](https://carpentries.github.io/lesson-example/) to learn more about how to use our template).

Yes, "Using The Carpentries style version 9.5.3."  However, we have customized the template and are using a fairly old version of the template as our base.

5. If you answered "No" to either part of question 4, would you like our Curriculum Team to create a repository for you in The Carpentries Incubator?

N/A

6. If you answered "Yes" to both parts of question 4, would you like to transfer your repository to The Carpentries Incubator? You will have [Admin access][access-levels] to the repository.

Yes

7. If you answered "Yes" to either question 5 or 6, list the GitHub handles for people who should have [Maintain access][access-levels] to your lesson. If you don't know how to answer this question, don't worry! We can always add collaborators later.

Can we please call the team `ml4bio-maintainers`?  Collaborators are:
- @cmilica
- @csmagnano

8. Any other information you would like us to have or questions you have for us?

Previously discussed with @tobyhodges on a video call.

Before transferring, I would like to plan how to set up automatic redirects from the existing https://gitter-lab.github.io/ URLs to the new GitHub Pages URLs.  We have other resources that link to these HTML pages and past participants have bookmarked those URLs.

Thank you for sharing your lesson with The Carpentries community!

[access-levels]: https://help.github.com/en/articles/repository-permission-levels-for-an-organization
[coc]: https://docs.carpentries.org/topic_folders/policies/code-of-conduct.html#code-of-conduct-summary-view
[lesson-template]: https://github.com/carpentries/styles

**Comment by tobyhodges** [comment]
Thank you for submitting this lesson to The Carpentries Incubator, @agitter! I'm delighted to see this here.

You should have received an invitation to join a Team of maintainers for the lesson, as part of the carpentries-incubator organisation on GitHub. After you have accepted that invitation, you should be able to initiate the transfer into carpentries-incubator, via the repository Settings. After that, I will make sure that you all have the appropriate access to the repository ("Admin" for @agitter, "Maintain" for @cmilica and @csmagnano).

Regarding the redirects, take a look at [the repository we recently set up to redirect people to a new URL](https://github.com/carpentrieslab/python-aos-lesson) for the Python for Atmosphere and Ocean Scientists lesson in The Carpentries Lab. This consists of two files: a README file to point people to the new GitHub repository, and an `index.md` which should automatically redirect visitors from the old lesson pages URL to the new location.

If you have any trouble with the redirect, the repo transfer, or anything else about the process, please post back here and I will be happy to try to help. I will leave this issue open until the transfer has been completed.

To keep in touch with The Carpentries curriculum community you may find it useful to join [the `incubator-developers` list on The Carpentries TopicBox](https://carpentries.topicbox.com/groups/incubator-developers) and/or [the `lesson-dev` channel on The Carpentries Slack](https://swcarpentry.slack.com/archives/C3KUTT5V3). Joining the Slack channel requires that you first [become a member of that workspace](https://swc-slack-invite.herokuapp.com/).

**Comment by agitter** [comment]
I completed the transfer.  @tobyhodges can you please check my permission levels on the https://github.com/carpentries-incubator/ml4bio-workshop.  I don't believe I have admin access.

Can you also please help me with the maintainers team?  https://github.com/orgs/carpentries-incubator/teams/ml-bio-maintainers/ only has you and me as members.  Can you please add @cmilica and @csmagnano too?

**Comment by agitter** [comment]
Thanks for the pointer to https://github.com/carpentrieslab/python-aos-lesson  I used that to create a new readme and redirect index in https://github.com/gitter-lab/ml-bio-workshop.  I also created a 404.html based on the GitHub Pages 404 page.

> Before transferring, I would like to plan how to set up automatic redirects from the existing https://gitter-lab.github.io/ URLs to the new GitHub Pages URLs. We have other resources that link to these HTML pages and past participants have bookmarked those URLs.

I was able to write a script to accomplish what I wanted.  It takes a manually-generated list of HTML pages in the old repository as input and then writes a redirect page for each of them.

**Comment by tobyhodges** [comment]
> I completed the transfer. @tobyhodges can you please check my permission levels on the https://github.com/carpentries-incubator/ml4bio-workshop. I don't believe I have admin access.

Thanks @agitter! I am very pleased to see this lesson join the Incubator. I have added you as Admin on the lesson repository now.

> Can you also please help me with the maintainers team? https://github.com/orgs/carpentries-incubator/teams/ml-bio-maintainers/ only has you and me as members. Can you please add @cmilica and @csmagnano too?

They should both have a invitation to accept, to join the Team. Let me know if you would like me to re-send either of those invitations. I have also now made you a Maintainer of that Maintainer Team (confusing, I know!), so you should be able to invite others to join if and when you need to do so. 

> I was able to write a script to accomplish what I wanted. It takes a manually-generated list of HTML pages in the old repository as input and then writes a redirect page for each of them.

Nice!

I will close this issue now that the transfer is complete, but I'll still receive notifications so feel free to post here again if you have any further questions.

**Comment by agitter** [comment]
Thanks @tobyhodges.  I cleaned up broken links in the transferred repository.  Everything looks good now.

I would like to update to the latest lesson template at some point and created a new issue for that (https://github.com/carpentries-incubator/ml4bio-workshop/issues/120).  I may tag you there if I get stuck or run into a difficult merge conflict.
